cmd/juju/caas/aks.go

```diff
@@ -0,0 +1,287 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package caas
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/juju/cmd"
+	"github.com/juju/collections/set"
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/caas/kubernetes/clientconfig"
+	"github.com/juju/juju/cmd/juju/interact"
+)
+
+type aks struct {
+	CommandRunner
+	azExecName string
+}
+
+func newAKSCluster() k8sCluster {
+	return &aks{CommandRunner: &defaultRunner{}}
+}
+
+func (a *aks) cloud() string {
+	return "aks"
+}
+
+func (a *aks) ensureExecutable() error {
+	if a.azExecName != "" {
+		return nil
+	}
+
+	cmd := []string{"which", "az"}
+	err := collapseRunError(runCommand(a, cmd, ""))
+	if err != nil {
+		cmd = []string{"which", "azure-cli"}
+		err := collapseRunError(runCommand(a, cmd, ""))
+		errAnnotationMessage := "neither az or azure-cli (version > 2.0.52) found. Please install, login, and try again"
```

Comment:
Instead of "Please install"....

Tell them which one to install and how, eg "snap install foo" or "apt install foo" or whatever

Reply:
hah, are we ok with having urls to web pages for install? As it stands right now the snap is not suitable to be used (not in stable and an older version.) and the deb requires addition of a ppa + key etc. as per https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-apt?view=azure-cli-latest